### PR TITLE
Add support for ctrl+Fn keys under rxvt

### PR DESCRIPTION
**Pull request by davep**
Found during #3440, this adds support for <kbd>ctrl</kbd>+<kbd>Fn</kbd> and <kbd>ctrl</kbd>+<kbd>sh</kbd>+<kbd>Fn</kbd> keys within rxvt. Mainly discovered and tested on macOS/XQuartz, but I think this will enable this under GNU/Linux too (Parallels keeps forgetting my GNU/Linux VMs for reasons I can't fathom -- I should probably just set up [UTM](https://mac.getutm.app/) for GNU/Linux like I have on my own machine[^1]).

See #3735 (and by extension #3440).

[^1]: I've since done that and tested it and sure enough I'm seeing the same thing on GNU/Linux too (urxvt v9.30); the unknown sequences are all the ones covered here.

### CODE REVIEW

**Comment by davep** [comment]
Also one that might interest @TomJGooding (I'd have added them as a reviewer too but GitHub says no).

**Comment by TomJGooding** [comment]
Thanks Dave! Here's a screenshot after a quick run through <kbd>Ctrl</kbd>+<kbd>Fn</kbd> and <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Fn</kbd> combinations:

image

Did you also find there's a couple of combinations that emit the same sequence? Then <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>F3</kbd> is F13 rather than F15 and so on?

```json
        {
            "key": "\\x1b[23^",
            "notes": "Ctrl+F11 AND Ctrl+Shift+F1"
        },
        {
            "key": "\\x1b[24^",
            "notes": "Ctrl+F12 AND Ctrl+Shift+F2"
        },
```

**Comment by davep** [comment]
Note the edit from the email version I just read: so we're good that that's just an oddity of how rxvt works?

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Widgets can now have an ALLOW_CHILDREN (bool) classvar to disallow adding children to a widget https://github.com/Textualize/textual/pull/3758
 - Added the ability to set the `label` property of a `Checkbox` https://github.com/Textualize/textual/pull/3765
 - Added the ability to set the `label` property of a `RadioButton` https://github.com/Textualize/textual/pull/3765
+- Added support for Ctrl+Fn and Ctrl+Shift+Fn keys in urxvt https://github.com/Textualize/textual/pull/3737
+- Added support for various modified edit and navigation keys in urxvt https://github.com/Textualize/textual/pull/3739
```
Comment (rodrigogiraoserrao) [inline CHANGELOG.md:25]: Which ones? Are these also the Ctrl + Fn keys added below?

**Review by rodrigogiraoserrao — APPROVED** [review]
I'm trusting you and Tom that the keys do work under rxvt.
I mashed some keys on my end and it doesn't look like you broke everything.

File: CHANGELOG.md (hunk 1 above)
Comment (davep) [reply CHANGELOG.md:25]: https://github.com/Textualize/textual/pull/3737/commits/a5650f73c4d5d9465cecb9b7b79aa3c9e7517619

**Comment by davep** [comment]
> I mashed some keys on my end and it doesn't look like you broke everything.

It's this level of quality testing and review that makes us such a formidable team.

**Pull request merged**